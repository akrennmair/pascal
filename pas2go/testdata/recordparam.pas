program test;

type foo = record
		b : integer;
		c : real;
	end;
	bar = record
		d : string;
		e : foo;
	end;

var y : bar;

procedure quux(x : bar);
begin
	x.d := 'hello';
	x.e.b := 42;
	x.e.c := 3.1415;
	writeln(x.d, x.e.b, x.e.c)
end;

begin
	quux(y)
end.
