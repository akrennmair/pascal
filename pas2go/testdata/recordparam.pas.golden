package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	type (
		foo struct {
			b int
			c float64
		}
		bar struct {
			d string
			e foo
		}
	)

	var (
		y bar
	)
	_ = y

	var quux func(x bar)
	quux = func(x bar) {
		x.d = "hello"
		x.e.b = 42
		x.e.c = 3.1415e0
		system.Writeln(x.d, x.e.b, x.e.c)
		return
	}

	quux(y)
}
